diff -up ./data/Makefile.am.ori ./data/Makefile.am
--- ./data/Makefile.am.ori	2013-04-29 19:28:57.000000000 +0200
+++ ./data/Makefile.am	2018-05-31 14:33:50.164626183 +0200
@@ -36,6 +36,11 @@ pkgconfig_DATA = polkit-gobject-1.pc pol
 
 # ----------------------------------------------------------------------------------------------------
 
+itsdir = $(datadir)/gettext/its
+its_DATA = polkit.loc polkit.its
+
+# ----------------------------------------------------------------------------------------------------
+
 systemdservice_in_files = polkit.service.in
 
 if HAVE_SYSTEMD
diff -up ./data/polkit.its.ori ./data/polkit.its
--- ./data/polkit.its.ori	2018-05-31 14:33:50.164626183 +0200
+++ ./data/polkit.its	2018-05-31 14:33:50.164626183 +0200
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<its:rules xmlns:its="http://www.w3.org/2005/11/its"
+           version="2.0">
+  <its:translateRule selector="//*" translate="no"/>
+  <its:translateRule selector="//action/description |
+                               //action/message"
+                     translate="yes"/>
+</its:rules>
diff -up ./data/polkit.loc.ori ./data/polkit.loc
--- ./data/polkit.loc.ori	2018-05-31 14:33:50.165626179 +0200
+++ ./data/polkit.loc	2018-05-31 14:33:50.164626183 +0200
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
+<locatingRules>
+  <locatingRule name="polkit policy" pattern="*.policy">
+    <documentRule localName="policyconfig" target="polkit.its"/>
+  </locatingRule>
+</locatingRules>
